frontend/ui/widget/opdsbrowser.lua

```diff
@@ -479,10 +492,7 @@ function OPDSBrowser:downloadFile(item, format, remote_url)
 
     if c == 200 then
         DEBUG("file downloaded to", local_path)
-        UIManager:show(InfoMessage:new{
-            text = _("File saved to:\n") .. local_path,
-            timeout = 3,
-        })
+        self:openBook(local_path)
```

Comment:
I think a better way to handle this is use the same callback pattern like `close_callback`.

You can add a new `file_downloaded_callback` to OPDSBrowser widget and change this line to:

```lua
if self.file_downloaded_callback then
    self.file_downloaded_callback(local_path)
end
```

Then everything else can be moved to opdscatalog.lua:

```lua
local opds_browser = OPDSBrowser:new{
    ...
    file_downloaded_callback = function(downloaded_file) 
        UIManager:show(ConfirmBox:new{
            text = T(_("File saved to:\n %1\nWould you like to read the downloaded book now?"),
                     downloaded_file),
            ok_callback = function()
                ReaderUI:showReader(downloaded_file)
                self:onClose()
            end
        }
    end,
    ...
}		      
```